### PR TITLE
Android 12 build support.

### PR DESCRIPTION
**Pull request by ipdev99**
Android 12 and 12L build support.

Change in the installer script is cosmetic.
_Since both 12 and 12L use `ro.build.version.release=12`
This will display SDK32 as 12L instead of 12._

### CODE REVIEW

**Comment by ipdev99** [comment]
Since Android 9 (SDK28) Google stopped incremental builds.
_I could not find the article (might have been a page that Google changed/deleted/updated).
I may have missed bringing back incremental builds. :upside_down_face:_

Starting from SDK 28, each new level would be a new release level.
_Until 12L :upside_down_face:_

The build branch was updated to 12.1 when 12L was released but..

Platform codenames, versions, API levels, and NDK releases - [source.android](https://source.android.com/setup/start/build-numbers) has been updated a few times.
_When I started on 12L builds, it was not listed. Then it was listed as Android 12L when release.
Android 12L is now listed as version 12.1_

Android 12 (31) and 12L (32) still share the 12 release tag.
_12L still shows (May 2022 factory Pixel releases) as 12, not 12.1 or 12L_

I am not sure how to name it. :thinking:
_Stay with 12 and 12L or switch to 12.0 and 12.1_

Cheers. :cowboy_hat_face:

**Comment by flame-0** [comment]
I am wondering if I can use this now?

**Comment by josephcsible** [comment]
Yes, via test builds posted at https://forum.xda-developers.com/t/gapps-daily-open-gapps-for-android-all-android-versions-devices.3098071/post-86595145

**Comment by mfonville** [comment]
Thanks for your PR!

@nezorflame would the updated naming scheme with 12 and 12L work well with our website API for downloads?

**Comment by nezorflame** [comment]
Thanks, I'll check this.

**Comment by mfonville** [comment]
@nezorflame were you able to test the website compatibility?

**Comment by khalidtarek** [comment]
Is this branch supporting Android 12 and 12L for x86_64? or for ARM only?
I am trying to add Gapps to the latest windows WSA using [WSAGAScript](https://github.com/WSA-Community/WSAGAScript) and when I check out this branch and perform 
./download_sources.sh --shallow x86_64
then  make x86_64-32-pico 
when I use the result image with the WSAGAScript it is causing the WSA to stuck at booting. so I am just checking if this should work for x86_64 architecture or only for arm

**Comment by AshiVered** [comment]
There is an update? I need Open Gapps to Android 12 in his Stock ROM no GAPS ... I can to spoofing fingerprint, but I need OpenGaps ... thank you very much!

**Comment by Cortexian** [comment]
Trying to include this in Windows Subsystem for Android and this is an issue. Unable to login, possible due to this.

**Review by nezorflame — APPROVED** [review]

**Comment by nezorflame** [comment]
Think we can merge this now and enable in the build server settings.

**Comment by nezorflame** [comment]
[Supported ](https://github.com/opengapps/package-api/pull/5)the change in the website's API as well

**Comment by PeterNjeim** [comment]
Why was this PR closed?

**Review by Howard20181 — CHANGES_REQUESTED** [review]

File: scripts/build_gapps.sh [hunk 1]
```diff
@@ -146,6 +147,11 @@ api27hack       #only here for completeness
 api28hack       #only on 9.0+ we also include Actions Services, AndroidPlatformServices, Data Transfer Tool, Markup, Sounds
 api29hack       #only on 10.0+ we also include Actions Services with Pixel Launcher and TrichromeLibrary with Chrome and Webview
 api30hack       #only on 11.0+ we also include Actions Services with Pixel Launcher and TrichromeLibrary with Chrome and Webview
+<<<<<<< HEAD
+=======
+api31hack       #only on 12 - Don't Panic
+api32hack       #only on 12L - Don't Panic
+>>>>>>> 80e79a7 (Android 12 build support.)
```
[the remark this answers is not in the document]
Comment (ipdev99) [reply scripts/build_gapps.sh]: I missed thatn one. :upside_down_face:

Fixed. :cowboy_hat_face:

**Comment by ThatBlockyPenguin** [comment]
Just wondering how near this is to being merged - need this for OpenGApps on WSA

**Comment by s1204IT** [comment]
Once this is merged, the [MindTheGappsBuilder](//github.com/s1204IT/MindTheGappsBuilder) I created will no longer be needed as the WSA's main GApps.

@mfonville Everyone is waiting for your reaction.

**Comment by mfonville** [comment]
Sorry for my delay, was not available because of illness (combined with already full schedule).

Thanks for the rebase!

**Comment by nezorflame** [comment]
@mfonville will you be able to enable the builds?

**Comment by amrf04** [comment]
Would also like to know when Android 12 will be available

**Comment by s1204IT** [comment]
Until OpenGApps 12/12L is available, you will need to use other services such as MindTheGapps.
Especially users of WSA and LineageOS 19 and later are very keen on this.

**Comment by PeterNjeim** [comment]
> Until OpenGApps 12/12L is available, you will need to use other services such as MindTheGapps.
> Especially users of WSA and LineageOS 19 and later are very keen on this.

Not true, you can build it yourself and it works. It's been working for several months for in [my repository](https://github.com/PeterNjeim/MagiskOnWSA), where I offer Full OpenGApps support. Understandably, making the API offer the builds is better and is the goal, however claiming that users "need" to use something else right now is simply not true

**Comment by s1204IT** [comment]
@PeterNjeim You advertise your repository too much in various places.
I don't like it.
Each one of us can choose any means we like.

**Comment by PeterNjeim** [comment]
> @PeterNjeim You advertise your repository too much in various places.
> I don't like it.
> Each one of us can choose any means we like.

That isn't a response to what I said. You also reacted negatively to one of comments via an emoji and I asked why, and you never replied. I don't like that.

And I'm not advertising, I'm giving concrete examples of how you can build it yourself. Please boil down my rebuttal to your statement to my mention of my repository, my point still stands, and I stand by it

**Comment by nezorflame** [comment]
While true, we still need to support those who don't want to build them manually.
@mfonville any progress on the buildserver?

**Comment by slo617** [comment]
Of cause there is no law that forces anybody to do anything in open source projects. Everything is voluntary. (Never forget about that.)
Thank you for your work!
It would be very nice for the less-skilled part of the community and also for those how are busy with other projects if someone could update the build-servers.
Pretty please! ;)

**Comment by nezorflame** [comment]
@slo617 yes, that's what I meant, @mfonville is in charge of the servers.

**Comment by mfonville** [comment]
It is a bit of murphy's law at the moment. I was busy with de-odexing recent Android images to make sure we got the original and up-to-date sourced APKs. And already had little time for that available. Meanwhile the family is ill, so I am barely able to do any work at all, and on top of that is the physical machine that hosts the build-server offline at the moment, and its admin is abroad so he cannot look into it at the moment.

So I can only ask for some more patience...

**Comment by nezorflame** [comment]
Sure thing, thank you for doing this and best wishes to the family ❤️

**Comment by bootsie123** [comment]
I hope you're doing better! Just curious to know if there are any updates on this

**Comment by nezorflame** [comment]
We're going to deprecate our current buildserver and make everything
in-house from our main server.

Work is in progress for now so stay tuned for updates.

_____________________________

Best regards,
Ilya Danilkin

**Comment by lorand-horvath** [comment]
@nezorflame @mfonville I hope you're doing well! Just wanted to ask how is it going with the 12L builds... are you still having issues with the build server?

**Pull request merged**